src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateStorageSession.cs

```diff
@@ -56,7 +56,7 @@ protected override void Setup(FeatureConfigurationContext context)
             });
 
             context.Services.AddScoped<ICompletableSynchronizedStorageSession, NHibernateSynchronizedStorageSession>();
-            context.Services.AddScoped(sb => sb.GetRequiredService<ICompletableSynchronizedStorageSession>().StorageSession());
+            context.Services.AddScoped(sp => (sp.GetService<ISynchronizedStorageSession>() as INHibernateStorageSession)!);
```

Comment:
If I understand https://github.com/Particular/NServiceBus.NHibernate/blob/master/src/NServiceBus.NHibernate/SynchronizedStorageSessionExtensions.cs#LL27C25-L27C51 correctly, the `ICompletableSynchronizedStorageSession` implementation doesn't directly implement `INHibernateStorageSession` but `INHibernateStorageSession` instead which is accessed to resolve the instance. The casting here would try to cast the session provider into a session instead.

Reply:
Ah yeah I remember now. Will fix